### ISSUE TITLE
NotImplementedError while trying to train YoloV8

### ISSUE THREAD

**Issue by LeonLewiz** [ultralytics/ultralytics#17704]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

![изображение_2024-11-22_013449794](https://github.com/user-attachments/assets/48f5c75f-ff3e-4813-9b5a-77e15907cb6f)

### Environment

OS                  Windows-10-10.0.19045-SP0
Environment         Windows
Python              3.11.6
Install             pip
RAM                 31.93 GB
Disk                222.4/238.4 GB
CPU                 AMD Ryzen 5 5600 6-Core Processor
CPU count           12
GPU                 NVIDIA GeForce RTX 4070, 12281MiB
GPU count           1
CUDA                12.4

numpy               ✅ 1.26.4>=1.23.0
numpy               ✅ 1.26.4<2.0.0; sys_platform == "darwin"
matplotlib          ✅ 3.9.2>=3.3.0
opencv-python       ✅ 4.9.0.80>=4.6.0
pillow              ✅ 10.4.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.14.1>=1.4.1
torch               ✅ 2.5.1+cu124>=1.8.0
torch               ✅ 2.5.1+cu124!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.20.1>=0.9.0
tqdm                ✅ 4.67.0>=4.64.0
psutil              ✅ 6.1.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.3>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.11>=2.0.0

### Minimal Reproducible Example

from ultralytics import YOLO
model = YOLO("yolov8m.pt")
device = 'cuda'
results = model.train(data="ebatnyas/data.yaml")

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @LeonLewiz, thank you for your interest in Ultralytics 🚀! This is an automated response, and an Ultralytics engineer will assist you soon.

We recommend checking out the [Docs](https://docs.ultralytics.com) for guidance, especially if you're new to our platform. You'll find many useful [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples that may address your issue.

If this is a 🐛 Bug Report, it looks like you've provided a screenshot and some environment details, which is great! To help us investigate this further, please ensure you include a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) if you haven't done so already. This will assist our team in debugging.

Here's something you can try in the meantime to see if your issue resolves:

### Upgrade

Ensure you're using the latest version of the `ultralytics` package, along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml). It needs to be run in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

### Environments

YOLO is compatible with these verified environments that come pre-installed with all dependencies like [CUDA](https://developer.nvidia.com/cuda), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Join our community for support and discussions! For real-time chat, head to [Discord](https://ultralytics.com/discord) 🎧. Prefer forums? Visit [Discourse](https://community.ultralytics.com) or our [Subreddit](https://reddit.com/r/ultralytics).

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

Our CI pipeline checks all YOLO modes and tasks across different platforms. If this badge is green, all tests are passing. You can track updates and check the system status at [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule).

Thanks again for reaching out! An engineer will be in touch soon to help you further 🤖.

**Comment by Y-T-G** [comment]
Uninstall torch and torchvision
`pip uninstall -y torch torchvision`

And reinstall using the command in the website:
https://pytorch.org/get-started/locally/
